src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -152,7 +156,16 @@ public void CounterPayloadReceived(string providerName, ICounterPayload payload,
 
         public void Stop()
         {
-            // Nothing to do here.
+            if (initialized)
+            {
+                var row = maxRow;
+
+                if (row > -1)
+                {
+                    Console.SetCursorPosition(0, row);
+                    Console.WriteLine();
+                }
+            }
```

Comment:
`Stop` can be called while `OnCounterPayloadReceived` is executing.  This means we might need to put synchronization around `maxRows`.

Reply:
What kind of synchronization do you have in mind? Making `CounterPayloadReceived` and `Stop` mutually exclusive with a lock, or just volatile access around `maxRows`?

Reply:
The other exporters use mutual exclusion, e.g.,
https://github.com/dotnet/diagnostics/blob/8e5123a15b40902abd9813927ce09ddd8d2c688a/src/Tools/dotnet-counters/Exporters/JSONExporter.cs#L59-L88

Reply:
Alright 👍